### PR TITLE
Patient History Enhancement

### LINKED ISSUE #803: Patient History Enhancement
5 enhancement points in patient history page:

klik tombol ini hingga berwarna hijau untuk memunculkan detail pertanyaan kontak (tracing kontak) (7)

### PR DESCRIPTION
**Pull request by yussin-hidayat**
close #803

### CODE REVIEW

**Comment by vincent-karuri** [comment]
@yussin-hidayat isn't this issue about displaying the date that's already on the visits page on the patient history page?

**Comment by yussin-hidayat** [comment]
> @yussin-hidayat isn't this issue about displaying the date that's already on the visits page on the patient history page?

Yes. And also about showing "date conducted" field.

**Comment by vincent-karuri** [comment]
What's the difference between `date conducted` and `visit date`?

**Comment by yussin-hidayat** [comment]
> What's the difference between `date conducted` and `visit date`?

Date conducted is the date when x-ray scan performed.

**Comment by vincent-karuri** [comment]
> > What's the difference between `date conducted` and `visit date`?
> 
> Date conducted is the date when x-ray scan performed.

And this can be done on a different day from when the form is submitted?

**Comment by yussin-hidayat** [comment]
> > > What's the difference between `date conducted` and `visit date`?
> > 
> > 
> > Date conducted is the date when x-ray scan performed.
> 
> And this can be done on a different day from when the form is submitted?

Yes.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/CovidPatientHistoryActivity.java [hunk 1]
```diff
@@ -43,6 +43,10 @@ protected void onCreate(Bundle savedInstanceState) {
         currPatient = getIntent().getParcelableExtra(Constants.FormFields.PATIENT);
         presenter = new CovidPatientHistoryActivityPresenter(this);
 
+        TextView tv = findViewById(R.id.visit_date_label);
+        String visitDateLabel = getString(R.string.visit_date_label);
+        tv.setText(visitDateLabel + ": " + getIntent().getStringExtra(Constants.FormFields.PATIENT_VISIT_DATE));
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/CovidPatientHistoryActivity.java:48]: We can move `getIntent().getStringExtra(Constants.FormFields.PATIENT_VISIT_DATE))` to a global variable, it's used twice in the file.

File: rdt-open-reader/src/covidIndonesia/res/layout/activity_covid_patient_history.xml [hunk 2]
```diff
@@ -1,51 +1,68 @@
...
-            android:layout_height="match_parent"
+    <LinearLayout
             android:layout_width="match_parent"
+            android:layout_height="match_parent"
             android:orientation="vertical">
 
-            <include
-                android:id="@+id/patient_symptoms_history_section"
-                layout="@layout/patient_history_section"
+        <androidx.core.widget.NestedScrollView
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/res/layout/activity_covid_patient_history.xml:10]: Why not add the date within this `NestedScrollView`?

Comment (yussin-hidayat) [reply rdt-open-reader/src/covidIndonesia/res/layout/activity_covid_patient_history.xml:10]: Ellya asked to make it fixed.

Comment (vincent-karuri) [reply rdt-open-reader/src/covidIndonesia/res/layout/activity_covid_patient_history.xml:10]: What issue is being addressed by the change?

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/CovidPatientHistoryActivity.java (hunk 1 above)
Comment (yussin-hidayat) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/CovidPatientHistoryActivity.java]: You mean instance variable? Done.

File: rdt-open-reader/src/covidIndonesia/res/layout/activity_covid_patient_history.xml (hunk 2 above)
Comment (yussin-hidayat) [reply rdt-open-reader/src/covidIndonesia/res/layout/activity_covid_patient_history.xml:10]: I mean "fixed layout", not scrolling when you scroll on scrollview. :D

Comment (vincent-karuri) [reply rdt-open-reader/src/covidIndonesia/res/layout/activity_covid_patient_history.xml:10]: like the date should be fixed (not move when scrolling)?

Comment (yussin-hidayat) [reply rdt-open-reader/src/covidIndonesia/res/layout/activity_covid_patient_history.xml:10]: Yes.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/CovidPatientHistoryActivity.java [hunk 3]
```diff
@@ -42,6 +43,11 @@ protected void onCreate(Bundle savedInstanceState) {
 
         currPatient = getIntent().getParcelableExtra(Constants.FormFields.PATIENT);
         presenter = new CovidPatientHistoryActivityPresenter(this);
+        patientVisitDate = getIntent().getStringExtra(Constants.FormFields.PATIENT_VISIT_DATE);
+
+        TextView tv = findViewById(R.id.visit_date_label);
+        String visitDateLabel = getString(R.string.visit_date_label);
+        tv.setText(visitDateLabel + ": " + patientVisitDate);
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/CovidPatientHistoryActivity.java:50]: Could this move to `populatePatientHistory()` function?

Comment (yussin-hidayat) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/CovidPatientHistoryActivity.java:50]: Sure.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/CovidPatientHistoryActivity.java [hunk 4]
```diff
@@ -66,6 +68,10 @@ private void populatePatientHistory() {
         for (Map.Entry<Integer, String> sectionIdAndEvent : getPatientHistorySectionsMap().entrySet()) {
             fetchAndPopulateHistory(sectionIdAndEvent.getKey(), sectionIdAndEvent.getValue());
         }
+
+        TextView tv = findViewById(R.id.visit_date_label);
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/CovidPatientHistoryActivity.java:72]: Could this be renamed to `tvVisitDate`?

**Review by vincent-karuri — APPROVED** [review]

**Pull request merged**